src/main/java/com/google/sps/servlets/UserCardsServlet.java

```diff
@@ -75,36 +83,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     if (userService.isUserLoggedIn()) {
 
-        // if blobKey is null, user did not add an image
         String folderKey = request.getParameter("folderKey");
-        String blobKey = getBlobKeyfromBlobstore(request, "image");
         String labels = request.getParameter("labels");
         String fromLang = request.getParameter("fromLang");
         String toLang = request.getParameter("toLang");
-        String textNotTranslated = request.getParameter("textNotTranslated");
-        String textTranslated = translateText(textNotTranslated, toLang);
+        String rawText = request.getParameter("rawText");
+        String textTranslated = request.getParameter("translatedText");
+        String blobKey = getBlobKey(request);
 
-        Card card = new Card(blobKey, labels, fromLang, toLang, textNotTranslated, textTranslated);
+        Card card = new Card(blobKey, labels, fromLang, toLang, rawText, textTranslated);
```

Comment:
At this point, I would suggest using a Builder pattern.

https://dzone.com/articles/design-patterns-the-builder-pattern

Reply:
This is a really cool design pattern. Thanks